### ISSUE TITLE
cordova:build does not stop when app build fails

### ISSUE THREAD

**Issue by adamniedzielski** [isleofcode/ember-cordova#73]
```
ember cordova:build --platform android

Build failed.
File: language-app/services/audio-player.js
The Broccoli Plugin: [Babel] failed with:
SyntaxError: language-app/services/audio-player.js: Unexpected token (20:37)

  _mediaError(error) {
    console.log(JSON.stringify(error);
  }
});

[...]

Symlinked dist/ to cordova/www

[...]

BUILD SUCCESSFUL
```

I believe that if application build fails then the dist directory should not be symlinked and Cordova CLI should not be invoked at all. What do you think?

**Comment by alexblom** [comment]
Agree. In a newer branch (#76 if memory serves) we are actually dropping the symlink to a full copy.

Agree that this should otherwise stop the process, and will confirm it does in the next few releases. 

If you'd like to take an attempt, feel free as well. @alexblom on ember-slack if you need direction.

**Comment by adamniedzielski** [comment]
After looking into the code I believe that the root cause is explained by this comment https://github.com/isleofcode/ember-cordova/blob/74870c92273318910093d566d43f99681a894a6a/lib/tasks/bash.js#L17

But why are the next tasks executed if the promise never resolves? The answer is here https://github.com/isleofcode/ember-cordova/blob/74870c92273318910093d566d43f99681a894a6a/lib/commands/build.js#L55-L58 Instead of passing a reference to function we invoke the function which means that the code executes immediately and promises are not used.

**Comment by anulman** [comment]
Good catch & diagnosis @adamniedzielski. I have a branch that solves exactly this, and uses a common interface + pattern for Tasks to guide development moving forward—will wait on @alexblom for the merge update ;)

**Comment by alexblom** [comment]
This should be resolved as of 0.2.0
